### PR TITLE
COASTAL-460: Adds triggerDateMatching for `.nextDate` and `.nextDateRepeating`

### PR DESCRIPTION
**Pull request by rickpasetto**
Note: DateComponents already conforms to Codable

https://tidepool.atlassian.net/browse/COASTAL-460

### CODE REVIEW

File: TidepoolKit/Model/Datum/TAlertDatum.swift [hunk 1]
```diff
@@ -25,12 +25,15 @@ public class TAlertDatum: TDatum, Decodable {
...
         case repeating
+        case nextDate
+        case nextDateRepeating
     }
     
     public var name: String?
     public var priority: Priority?
     public var trigger: Trigger?
     public var triggerDelay: TimeInterval?
+    public var triggerDateMatching: DateComponents?
```
Comment (darinkrauss) [inline TidepoolKit/Model/Datum/TAlertDatum.swift]: I think we're going to need to create a TidepoolKit type that partially mirrors `DateComponents`. The reason for doing this is that the backend will need to be updated to support this new field and data type. However, I don't think it is reasonable to expect that the backend support every known option of `DateComponents` as there are a lot that we won't likely use (e.g. era, yearForWeekOfYear, isLeapMonth, ...). I think we need to explicitly state we support only certain fields and define a type here to capture that.

File: TidepoolKitTests/Model/Datum/TAlertDatumTests.swift [hunk 2]
```diff
@@ -10,11 +10,13 @@ import XCTest
 import TidepoolKit
 
 class TAlertDatumTests: XCTestCase {
+    static let noon = DateComponents(hour: 12, minute: 0)
```
Comment (darinkrauss) [inline TidepoolKitTests/Model/Datum/TAlertDatumTests.swift]: [nit] We should test something that includes all of the *supported* date components. (See previous comment.)

**Review by darinkrauss — CHANGES_REQUESTED** [review]
See comments. Plus, what is the plan for updating the backend to support these changes? If we merge this now it will entirely break uploading alerts and I don't think we want that regression.

File: TidepoolKit/Model/Datum/TAlertDatum.swift (hunk 1 above)
Comment (rickpasetto) [reply TidepoolKit/Model/Datum/TAlertDatum.swift]: @darinkrauss and I chatted and I've agreed to make a specific type that only supports `days`, `hours`, and `minutes` for both `LoopKit` and `TidepoolKit`.  Stay tuned for the changes...

**Comment by rickpasetto** [comment]
After discussion with @ps2, product and partner, decided to not go forward with this change.

**Pull request closed without merging**